engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/model/transformer/UserTaskTransformer.java

```diff
@@ -54,6 +63,41 @@ private void transformTaskDefinition(final JobWorkerProperties jobWorkerProperti
     jobWorkerProperties.setRetries(new StaticExpression("1"));
   }
 
+  private void transformAssignmentDefinition(
+      final UserTask element, final JobWorkerProperties jobWorkerProperties) {
+    final var assignmentDefinition =
+        element.getSingleExtensionElement(ZeebeAssignmentDefinition.class);
+    if (assignmentDefinition == null) {
+      return;
+    }
+    final var assignee = assignmentDefinition.getAssignee();
+    if (assignee != null && !assignee.isBlank()) {
+      jobWorkerProperties.setAssignee(expressionLanguage.parseExpression(assignee));
+    }
+    final var candidateGroups = assignmentDefinition.getCandidateGroups();
+    if (candidateGroups != null && !candidateGroups.isBlank()) {
+      final var candidateGroupsExpression = expressionLanguage.parseExpression(candidateGroups);
+      if (candidateGroupsExpression.isStatic()) {
+        // static candidateGroups must be in CSV format, but this is already checked by validator
+        jobWorkerProperties.setCandidateGroups(
+            ZeebeExpressionValidator.parseListOfCsv(candidateGroups)
```

Comment:
@saig0 I had a look at this, and moved both expression transformations into a utility class. I'm not sure about the package and naming of this class because Transformer has a very specific meaning in the engine. WDYT? Any ideas on how to improve this?

Reply:
I'm okay with the names :+1: No other suggestion